controllers/envoy/revisions.go

```diff
@@ -27,119 +31,113 @@ func (r *EnvoyConfigReconciler) ensureEnvoyConfigRevision(ctx context.Context,
 
 	// Get the list of revisions for the current version
 	ecrList := &envoyv1alpha1.EnvoyConfigRevisionList{}
-	selector, err := metav1.LabelSelectorAsSelector(&metav1.LabelSelector{
-		MatchLabels: map[string]string{nodeIDTag: ec.Spec.NodeID, versionTag: version},
-	})
-	if err != nil {
-		return newCacheError(UnknownError, "ensureEnvoyConfigRevision", err.Error())
-	}
-	err = r.Client.List(ctx, ecrList, &client.ListOptions{LabelSelector: selector})
-	if err != nil {
-		if err != nil {
-			return newCacheError(UnknownError, "ensureEnvoyConfigRevision", err.Error())
-		}
+	envoyAPI := string(ec.GetEnvoyAPIVersion())
+	if err := r.Client.List(ctx, ecrList, getRevisionListOptions(ec.Namespace, &ec.Spec.NodeID, &version, &envoyAPI)...); err != nil {
+		return NewControllerError(UnknownError, "ensureEnvoyConfigRevision", err.Error())
 	}
 
 	// Got wrong number of revisions
 	if len(ecrList.Items) > 1 {
-		return newCacheError(UnknownError, "ensureEnvoyConfigRevision", fmt.Sprintf("more than one revision exists for config version '%s', cannot reconcile", version))
+		return NewControllerError(UnknownError, "ensureEnvoyConfigRevision", fmt.Sprintf("more than one revision exists for config version '%s', cannot reconcile", version))
 	}
 
 	// Revision does not yet exists, create one
 	if len(ecrList.Items) == 0 {
 		// Create the revision for this config version
 		ecr := &envoyv1alpha1.EnvoyConfigRevision{
 			ObjectMeta: metav1.ObjectMeta{
-				Name:      fmt.Sprintf("%s-%s", ec.Spec.NodeID, version),
+				Name:      fmt.Sprintf("%s-%s-%s", ec.Spec.NodeID, envoyAPI, version),
```

Comment:
This was required to avoid name collisions between v2 and v3 EnvoyConfigRevisions.